onadata/libs/serializers/user_profile_serializer.py

```diff
@@ -147,11 +147,14 @@ def _get_params(self, attrs):
     def update(self, instance, validated_data):
         params = validated_data
 
-        # Check password if email is beig updated
+        # Check password if email is being updated
         if 'email' in params and 'password1' not in params:
             raise serializers.ValidationError(
                 _(u'Your password is required when updating your email '
                   u'address.'))
+        password = params.get("password1")
```

Comment:
if you move this above line 151, you can have the second conjuct in that conditional be `password`